src/activities/content/ContentWebLinkEntity.js

```diff
@@ -0,0 +1,164 @@
+import { Entity } from '../../es6/Entity';
+import { Actions, Classes } from '../../hypermedia-constants';
+import { performSirenAction } from '../../es6/SirenAction';
+
+/**
+ * ContentWebLinkEntity class representation of a d2l content-weblink entity.
+ */
+export class ContentWebLinkEntity extends Entity {
+
+	/**
+	 * @returns {string} Description html of the content-weblink item
+	 */
+	descriptionRichText() {
+		if (!this._entity) {
+			return null;
+		}
+		const subEntity = this._entity.getSubEntitiesByClass(Classes.content.description)[0];
+		if (!subEntity || !subEntity.properties) {
+			return null;
+		}
+		return subEntity.properties.html || '';
+	}
+
+	/**
+	 * @returns {string} Description text of the content-weblink item
+	 */
+	descriptionText() {
+		if (!this._entity) {
+			return null;
+		}
+		const subEntity = this._entity.getSubEntitiesByClass(Classes.content.description)[0];
+		if (!subEntity || !subEntity.properties) {
+			return null;
+		}
+		return subEntity.properties.text || '';
+	}
+
+	/**
+	 * @returns {boolean} external resource value (i.e. open in new tab or not)
+	 */
+	isExternalResource() {
+		if (!this._entity) {
+			return null;
+		}
+		return this._entity.hasClass(Classes.content.externalResource);
+	}
+
+	/**
+	 * @returns {string} Title of the content-weblink item
+	 */
+	title() {
+		return this._entity && this._entity.properties && this._entity.properties.title;
+	}
+
+	/**
+	 * @returns {string} Url of the content-weblink item
+	 */
+	url() {
+		return this._entity && this._entity.properties && this._entity.properties.url;
+	}
+
+	/**
+	 * Updates the web link to have the given description
+	 * @param {string} title rich text description to set on the web link
+	 */
+	async setWebLinkDescription(richText) {
+		if (!this._entity) {
+			return;
+		}
+		const action = this._entity.getActionByName(Actions.content.updateDescription);
+		if (!action) {
+			return;
+		}
+
+		const fields = [{ name: 'description', value: richText }];
+		await performSirenAction(this._token, action, fields);
```

Comment:
It is set up this way because of the way the state is setup in the activities repo and how the various save functions are executed. When we save things we have to call the appropriate save actions from the various components in the editor (see modules save stuff [here](https://github.com/BrightspaceHypermediaComponents/activities/blob/master/components/d2l-activity-editor/d2l-activity-content-editor/state/content.js#L76)) and await their response so we know they execute successfully. These save functions are consolidated [here](https://github.com/BrightspaceHypermediaComponents/activities/blob/master/components/d2l-activity-editor/mixins/d2l-activity-editor-container-mixin.js#L145) and executed from this mixin [here](https://github.com/BrightspaceHypermediaComponents/activities/blob/master/components/d2l-activity-editor/mixins/d2l-activity-editor-container-mixin.js#L149). This mixin expects async functions and that is why all these functions that perform actions need to `await`.